internal/xds/bootstrap/tlscreds/bundle_ext_test.go

```diff
@@ -224,6 +228,87 @@ func (s) TestCaReloading(t *testing.T) {
 	}
 }
 
+func (s) TestSPIFFEReloading(t *testing.T) {
+	clientSPIFFEBundle, err := os.ReadFile(testdata.Path("spiffe_end2end/client_spiffebundle.json"))
+	if err != nil {
+		t.Fatalf("Failed to read test SPIFFE bundle: %v", err)
+	}
+
+	// Write CA certs to a temporary file so that we can modify it later.
+	spiffePath := t.TempDir() + "/client_spiffe.json"
+	if err = os.WriteFile(spiffePath, clientSPIFFEBundle, 0644); err != nil {
+		t.Fatalf("Failed to write test SPIFFE Bundle %v: %v", clientSPIFFEBundle, err)
+	}
+	cfg := fmt.Sprintf(`{
+		"spiffe_trust_bundle_map_file": "%s",
+		"refresh_interval": ".01s"
+	}`, spiffePath)
+	tlsBundle, stop, err := tlscreds.NewBundle([]byte(cfg))
+	if err != nil {
+		t.Fatalf("Failed to create TLS bundle: %v", err)
+	}
+	defer stop()
+
+	serverCredentials := grpc.Creds(testutils.CreateServerTLSCredentialsCompatibleWithSPIFFE(t, tls.NoClientCert))
+	server := stubserver.StartTestService(t, nil, serverCredentials)
+	defer server.Stop()
+
+	conn, err := grpc.NewClient(
+		server.Address,
+		grpc.WithCredentialsBundle(tlsBundle),
+		grpc.WithAuthority("x.test.example.com"),
+	)
+	if err != nil {
+		t.Fatalf("Error dialing: %v", err)
```

Comment:
We are trying to move away from the term "dial" as much as possible for grpc channel creation. So, it might be better in this error message to say something like `"grpc.NewClient(%q) failed: %v", server.Address, err`.

Here and in other tests in this file.

Reply:
Done